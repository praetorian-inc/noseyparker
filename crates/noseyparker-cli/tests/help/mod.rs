//! Tests for Nosey Parker `help` functionality

use super::*;

#[test]
fn no_args() {
    assert_cmd_snapshot!(noseyparker!().assert().failure());
}

#[test]
fn help() {
    assert_cmd_snapshot!(noseyparker_success!("help"));
}

#[test]
fn help_short() {
    assert_cmd_snapshot!(noseyparker_success!("-h"));
}

#[test]
fn help_scan() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "scan"));
    });
}

#[test]
fn help_scan_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("scan", "-h"));
    });
}

#[test]
fn help_summarize() {
    assert_cmd_snapshot!(noseyparker_success!("help", "summarize"));
}

#[test]
fn help_summarize_short() {
    assert_cmd_snapshot!(noseyparker_success!("summarize", "-h"));
}

#[test]
fn help_report() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "report"));
    });
}

#[test]
fn help_report_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("report", "-h"));
    });
}

#[test]
fn help_datastore() {
    assert_cmd_snapshot!(noseyparker_success!("help", "datastore"));
}

#[test]
fn help_rules() {
    assert_cmd_snapshot!(noseyparker_success!("help", "rules"));
}

#[test]
fn help_github() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github"));
}

#[test]
fn help_github_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "-h"));
}

#[test]
fn help_github_repos() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github", "repos"));
}

#[test]
fn help_github_repos_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "repos", "-h"));
}

// #[test]
// fn version_short() {
//     assert_cmd_snapshot!(noseyparker_success!("-V"));
// }

#[test]
fn version_long() {
    with_settings!({
        filters => vec![
            (r"(?m)^(    [^:]+:[ \t]+).*$", r"$1<PLACEHOLDER>")
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("--version"));
    });
}

#[test]
fn version_command() {
    // there is no `version` command
    assert_cmd_snapshot!(noseyparker_failure!("version"));
}
