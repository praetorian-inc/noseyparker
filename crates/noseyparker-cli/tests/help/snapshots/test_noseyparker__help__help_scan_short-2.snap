---
source: crates/noseyparker-cli/tests/help/mod.rs
expression: stdout
---
Scan content for secrets

Usage: noseyparker scan [OPTIONS] --datastore <PATH> [INPUT]...

Options:
  -d, --datastore <PATH>        Use the specified datastore [env: NP_DATASTORE=]
  -j, --jobs <N>                Use N parallel scanning jobs [default: DEFAULT]
      --snippet-length <BYTES>  Include up to the specified number of bytes before and after each
                                match [default: 256]
  -r, --rules <PATH>            Use custom rules from the specified file or directory
  -h, --help                    Print help (see more with '--help')

Input Specifier Options:
  [INPUT]...                    Scan the specified file, directory, or local Git repository
      --git-url <URL>               Clone and scan the Git repository at the specified URL
      --github-organization <NAME>  Clone and scan accessible repositories belonging to the
                                    specified GitHub organization [aliases: github-org]
      --github-user <NAME>          Clone and scan accessible repositories belonging to the
                                    specified GitHub user
      --github-api-url <URL>        Use the specified URL for GitHub API access [default:
                                    https://api.github.com/] [aliases: api-url]
      --git-clone-mode <MODE>       Use the specified method for cloning Git repositories [default:
                                    bare] [possible values: bare, mirror]

Content Filtering Options:
      --max-file-size <MEGABYTES>  Do not scan files larger than the specified size [default: 100]
  -i, --ignore <FILE>              Use custom path-based ignore rules from the specified file

Global Options:
  -v, --verbose...       Enable verbose output
      --color <MODE>     Enable or disable colored output [default: auto] [possible values: auto,
                         never, always]
      --progress <MODE>  Enable or disable progress bars [default: auto] [possible values: auto,
                         never, always]

